packages/components/src/internal/url/URLResolver.ts

```diff
@@ -13,13 +13,173 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { fromJS, List, Map } from 'immutable';
+import { fromJS, List, Map, OrderedSet } from 'immutable';
 import { ActionURL, Experiment, Filter } from '@labkey/api';
 
-import { AppURL } from '../url/AppURL';
+import { AppURL, createProductUrl } from '../..';
 import { LineageLinkMetadata } from '../components/lineage/types';
+import { AppRouteResolver } from './AppURLResolver';
```

Comment:
nit: `AppRouteResolver` can be imported from `'../..'`.

Reply:
Taking from neighbor. Understood.

Reply:
In general at the moment it is better to import from your neighbor if you can. Basically:

- Import from your neighbors
- If you have to go up at all, go up to index if you can

I can try to import from '`../..` here but it is very possible it'll introduce new circular dependencies.